src/mocks/msw/handlers/userProfileHandlers.ts

```diff
@@ -81,4 +88,22 @@ export const userProfileHandlers = [
       return res(ctx.status(200), ctx.json(mockPaginatedEntityHeaders))
     },
   ),
+
+  /**
+   * Get a batch of user group headers
+   */
+  rest.get(
+    `${getEndpoint(
+      BackendDestinationEnum.REPO_ENDPOINT,
+    )}${USER_GROUP_HEADERS_BATCH}`,
+    async (req, res, ctx) => {
+      const ids = req.url.searchParams.get('ids')!.split(',')
+      const responsePage: UserGroupHeaderResponsePage = {
+        children: mockUserData
+          .filter(userData => ids.includes(userData.id.toString()))
+          .map(userData => userData.userGroupHeader),
+      }
+      return res(ctx.status(200), ctx.json(responsePage))
+    },
+  ),
```

Comment:
Mock fetching the userGroupHeader so we can determine if a principal is a user or team